README.md

```diff
@@ -5,757 +5,864 @@
+  designer: { name: "Homer", age: 39 },
+  name: "The Homer"
+});
+```
+
+`theHomerCar` object will have the following shape,
+
+```txt
++-------------------+           +----------------------+
+|                   |           |                      |       +--------------------+
+|  Microstate<Car>  |           |  Microstate<Person>  +-name--+                    +-concat()->
+|                   |           |                      |       | Microstate<String> +-set()->
+|                   +-designer--+                      |       |                    +-state: 'Homer'
+|                   |           |                      |       +--------------------+
+|                   |           |                      |
+|                   |           |                      |       +--------------------+
+|                   |           |                      +-age---+                    +-increment()->
+|                   |           |                      |       | Microstate<Number> +-decrement()->
+|                   |           |                      |       |                    +-set()->
+|                   |           |                      |       |                    +-state: 39
+|                   |           |                      |       +--------------------+
+|                   |           |                      |
+|                   |           |                      +-set()->
+|                   |           |                      +-state: Person<{ name: 'Homer', age: 39 }>
+|                   |           +----------------------+
+|                   |
+|                   |           +--------------------+
+|                   +-name------+                    +-concat()->
+|                   |           | Microstate<String> +-set()->
+|                   |           |                    +-state: 'The Homer'
+|                   |           +--------------------+
+|                   |
+|                   +-set()->
+|                   +-state: Car<{ designer: Person<{ name: 'Homer', age: 39 }> }, name: 'The Homer' }>
++-------------------+
 ```
 
-You can access transitions of composed states by using object property notation. Every time that you
-call a transition, you receive a new microstate and you start from the root of the original type.
+The property names are important when defining `class` types because they are used to reference composed microstates. You can use the object dot notation to access a composed microstate. Using the same example from above:
 
 ```js
-create(Person)
-  .age.increment()
-  .address.city.set('San Francisco')
-  .address.street.set('Market St').state;
-// => {
-//      name: 'Taras',
-//      age: 1,
-//      address: {
-//        street: 'Market St',
-//        number: 0,
-//        city: 'San Francisco'
-//      }
-//    }
+theHomerCar.designer.state;
+//> Person<{ name: 'Homer', age: 39 }>
+
+theHomerCar.designer.age.state;
+//> 39
+
+theHomerCar.name.state;
+//> The Homer
```

Comment:
Might also be worth showing the whole homer car state

> // or we can access the top level state from the top level microstate!
> theHomerCar.state;
> //> { name: "The Homer", designer: {name: 'Homer', age: 39} }

Reply:
Done